routes/sites.js

```diff
@@ -1,25 +1,84 @@
 'use strict';
 
-// const pool = require('../config/config').pool;
+const Promise = require('bluebird');
+const query = require('../lib/utils').query;
 
+/**
+ * Gets all sites.
+ *
+ * @param {Object} req The given request object
+ * @return {Promise} The promise
+ */
 function getSites(req) {
+  if (req.query && req.query.coach_id) {
+    return query('SELECT site_id, site_name, site_address FROM Acct NATURAL JOIN AcctToProgram NATURAL JOIN Program NATURAL JOIN Site WHERE acct_id = ' + req.query.coach_id);
+  }
 
+  return query('SELECT * FROM Site');
 }
 
+/**
+ * Creates a site.
+ *
+ * @param {Object} req The given request object
+ * @return {Promise} The promise
+ */
 function postSite(req) {
+  if (!req.body || !req.body.site_name || !req.body.site_address) {
+    return Promise.reject({
+      status: 406,
+      message: 'Must provide site\'s name, and address'
+    });
+  }
 
+  return query('INSERT INTO Site (site_name, site_address) VALUES ("' + req.body.site_name + '", "' + req.body.site_address + '")');
```

Comment:
Note: What if we already have a site named "Northeastern International Village" with address "1155 Tremont Street?" If you don't check that it exists before posting, it'll create a duplicate with a different ID. Add a test case to catch this (e.g. don't create site if it already exists). You don't need to throw an error, but maybe just do nothing and log a message. **Address before merging**